client/components/ReviewListItemText.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react';
+
+// add logice to show the "line" for key 0-3, but not 4
+
+function ReviewListItemText({ text, idx }) {
+  const [showMore, setShowMore] = useState(false);
+  if (idx === 4) {
+    if (showMore === false) {
+      if (text.length > 250) {
+        return (
+          <div>
+            <p className="reviewText">
+              {text.slice(0, 250)}
+              ...
+              <button
+                className="moreButton"
+                type="button"
+                onClick={() => setShowMore(true)}
+              >
+                +More
+              </button>
+            </p>
+          </div>
+        );
+      }
+      if (text.length <= 250) {
+        return (
+          <div>
+            <p className="reviewText">
+              {text}
+            </p>
+          </div>
+        );
+      }
+    } else {
+      return (
+        <div>
+          <p className="reviewText">{text}</p>
+        </div>
+      );
+    }
+  }
+  if (showMore === false) {
+    if (text.length > 250) {
+      return (
+        <div>
+          <p className="reviewText line">
+            {text.slice(0, 250)}
+            ...
+            <button
+              className="moreButton"
+              type="button"
+              onClick={() => setShowMore(true)}
+            >
+              +More
+            </button>
+          </p>
+        </div>
+      );
+    }
+    if (text.length <= 250) {
+      return (
+        <div>
+          <p className="reviewText line">
+            {text}
+          </p>
+        </div>
+      );
+    }
+  } else {
```

Comment:
I might be misunderstanding the logic. There are two if statements saying if "showMore === false" (lines 8 and 43) that seem to be saying the same thing, which is slice the first 250 chars, and add the +More button. If the text is less than 250 chars, show the entire text, else show all the entire text?

Reply:
Good point!  This section went thru a couple of iterations and I didn't optimize this as I added functionality.  I will fix it before I merge.  Thank you!

Reply:
I remember this now.  I need the second conditional because it is looking at the total length of the text, not just the text shown.  
the first 2 conditionals (>250 chars and <=250 chars) are only if the showMore state is false.  the last one is if the showMore state is true